e2e/consulcompat/run_ce_test.go

```diff
@@ -0,0 +1,262 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package consulcompat
+
+import (
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	consulapi "github.com/hashicorp/consul/api"
+	"github.com/hashicorp/nomad/helper/uuid"
+	"github.com/hashicorp/nomad/testutil"
+	"github.com/shoenig/test/must"
+)
+
+func testConsulBuildLegacy(t *testing.T, b build, baseDir string) {
+	t.Run("consul-legacy("+b.Version+")", func(t *testing.T) {
+		cStop, consulHTTPAddr, consulAPI := startConsul(t, b, baseDir, "")
+		t.Cleanup(func() { cStop() })
```

Comment:
The usual reason I do this is because it lets the caller control the order of shutdown when you have multiple cleanup functions. I'm not sure it matters in this particular case, but I haven't written the ENT-side tests yet.

Reply:
There's some grody stuff in the Consul start function though. Lemme see if I can clean this up a bit.

Reply:
Done!